ow_gazebo_plugins/src/BalovnevModelPlugin/BalovnevModelPlugin.cpp

```diff
@@ -213,23 +233,27 @@ void BalovnevModelPlugin::onModDiffVisualMsg(
     return;
   }
 
-  // get average height
-  float sum = 0.0;
-  for (auto y = 0; y < rows; ++y) {
-    for (auto x = 0; x < cols; ++x) {
-      // NOTE: pixel values should always be negative
-      sum += -image_handle->image.at<float>(y, x);
-    }
-  }
-  float depth = sum / (rows * cols);
+  // get max pixel value
+  // NOTE: pixel values should always be negative, so grab the minimum
+  double min_pixel;
+  cv::minMaxLoc(image_handle->image, &min_pixel);
+  m_moving_max_depth->addDatum(static_cast<float>(-min_pixel));
+
+  float depth = m_moving_max_depth->evaluate();
```

Comment:
Are these supposed to be cast/declared float? It looks like all of MovingMaxFilter members are doubles, so this would lose precision.

Reply:
Good catch! It should definitely just remain double. This may be a hold over from some previous iteration that used floats.

Reply:
I will investigate making the `depth` double as well tomorrow. I cannot recall why it's not right now, but it's the end of the day.